source_gen/test/type_checker_test.dart

```diff
@@ -173,37 +183,56 @@ void main() {
+            reason: '${checkNonPublic()} is not assignable from '
+                '${staticNonPublic.element.name}');
+      });
+    });
   }
 
   group('TypeChecker.forRuntime', () {
     commonTests(
-        checkIterable: () => const TypeChecker.fromRuntime(Iterable),
-        checkMap: () => const TypeChecker.fromRuntime(Map),
-        checkHashMap: () => const TypeChecker.fromRuntime(HashMap),
-        checkGenerator: () => const TypeChecker.fromRuntime(Generator),
-        checkGeneratorForAnnotation: () =>
-            const TypeChecker.fromRuntime(GeneratorForAnnotation));
+      checkIterable: () => const TypeChecker.fromRuntime(Iterable),
+      checkMap: () => const TypeChecker.fromRuntime(Map),
+      checkHashMap: () => const TypeChecker.fromRuntime(HashMap),
+      checkGenerator: () => const TypeChecker.fromRuntime(Generator),
+      checkGeneratorForAnnotation: () =>
+          const TypeChecker.fromRuntime(GeneratorForAnnotation),
+      checkNonPublic: () => const TypeChecker.fromRuntime(NonPublic),
+    );
   });
 
   group('TypeChecker.forStatic', () {
     commonTests(
-        checkIterable: () => staticIterableChecker,
-        checkMap: () => staticMapChecker,
-        checkHashMap: () => staticHashMapChecker,
-        checkGenerator: () => staticGeneratorChecker,
-        checkGeneratorForAnnotation: () => staticGeneratorForAnnotationChecker);
+      checkIterable: () => staticIterableChecker,
+      checkMap: () => staticMapChecker,
+      checkHashMap: () => staticHashMapChecker,
+      checkGenerator: () => staticGeneratorChecker,
+      checkGeneratorForAnnotation: () => staticGeneratorForAnnotationChecker,
+      checkNonPublic: () => staticNonPublicChecker,
+    );
   });
 
   group('TypeChecker.fromUrl', () {
     commonTests(
-        checkIterable: () => const TypeChecker.fromUrl('dart:core#Iterable'),
-        checkMap: () => const TypeChecker.fromUrl('dart:core#Map'),
-        checkHashMap: () =>
-            const TypeChecker.fromUrl('dart:collection#HashMap'),
-        checkGenerator: () => const TypeChecker.fromUrl(
-            'package:source_gen/src/generator.dart#Generator'),
-        checkGeneratorForAnnotation: () => const TypeChecker.fromUrl(
-            'package:source_gen/src/generator_for_annotation.dart#GeneratorForAnnotation'));
+      checkIterable: () => const TypeChecker.fromUrl('dart:core#Iterable'),
+      checkMap: () => const TypeChecker.fromUrl('dart:core#Map'),
+      checkHashMap: () => const TypeChecker.fromUrl('dart:collection#HashMap'),
+      checkGenerator: () => const TypeChecker.fromUrl(
+          'package:source_gen/src/generator.dart#Generator'),
+      checkGeneratorForAnnotation: () => const TypeChecker.fromUrl(
+          'package:source_gen/src/generator_for_annotation.dart#GeneratorForAnnotation'),
+      checkNonPublic: () => const TypeChecker.fromUrl(
+          'asset:source_gen/test/type_checker_test.dart#NonPublic'),
```

Comment:
This shows up differently internally so this test fails - internally we get a `google3://` not a `file://` URI.

If necessary we could patch a skip in or something...